granulate_utils/futures.py

```diff
@@ -0,0 +1,32 @@
+from concurrent.futures import Future, ThreadPoolExecutor, _base, as_completed
+from contextlib import contextmanager
+from typing import Callable, Collection, Generator, TypeVar
+
+T = TypeVar("T")
+
+
+@contextmanager
+def wrap_thread_pool(pool: ThreadPoolExecutor) -> Generator[ThreadPoolExecutor, None, None]:
+    try:
+        yield pool
+    except _base.TimeoutError as e:
+        raise TimeoutError(*e.args) from e
```

Comment:
What's this?

Reply:
For some reason they decided to implement their own version of TimeoutError, and only fixed it in python 3.11, so I translate their TimeoutError to the standard one:
<img width="801" alt="image" src="https://user-images.githubusercontent.com/3042955/226160101-ec909f61-90fb-4c26-9df6-8bb465938ea1.png">
I'll document it.